### PR TITLE
fix: Ensure WASM- and Skia-specific `WinRTFeatureConfiguration` is accessible

### LINKED ISSUE #13057: `WinRTFeatureConfiguration` for `ApplicationData` is not accessible
### Current behavior

_No response_

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #13057

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Previously these properties were wrapped in __SKIA__ and __WASM__ conditionals, which prevent them from being accessible via the reference API surface - which makes the impossible to be used outside of Uno solution.

## What is the new behavior?

- Using `__CROSSRUNTIME__` to make them accessible everywhere from real apps.

### <samp>🤖 Generated by Copilot at dc5fe02</samp>

This pull request refactors and enhances the feature configuration classes and files for Uno Platform applications. It moves the `WinRTFeatureConfiguration` class and its nested classes to a new folder `FeatureConfiguration`, and adds new classes for accessibility, array pool, file types, gesture recognizer, MIDI, network information, and web authentication broker features. It also enables more platform overrides for the application data paths.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs [hunk 1]
```diff
@@ -22,27 +22,28 @@ public static class Storage
...
 	[Flags]
 	public enum WasmPickerConfiguration
 	{
 		FileSystemAccessApi = 1,
 		DownloadUpload = 2,
 		FileSystemAccessApiWithFallback = FileSystemAccessApi | DownloadUpload,
 	}
-#endif
 }
+#endif
```
Comment (Youssef1313) [inline src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs:49]: Is this `#endif` incorrecly moved?

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs [hunk 2]
```diff
@@ -22,27 +22,28 @@ public static class Storage
 		{
 			public static class Pickers
 			{
-#if UNO_REFERENCE_API
+#if __CROSSRUNTIME__
```
Comment (Youssef1313) [inline src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs:25]: Both `__CROSSRUNTIME__` and `UNO_REFERENCE_API` are currently true for Wasm, Skia, and Reference.

Does this change have no impact, but for future when UNO_REFERENCE_API is properly fixed to mean reference only?

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.Focus.cs [hunk 3]
```diff
@@ -1,14 +1,15 @@
-﻿namespace Uno;
-
-partial class WinRTFeatureConfiguration
+﻿namespace Uno
 {
```
Comment (Youssef1313) [inline src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.Focus.cs:2]: Why is this moving from file-scoped namespaces to block namespaces?

**Review by Youssef1313 — APPROVED** [review]
LGTM

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs:25]: I mainly changed this for consistency, the `__CROSSRUNTIME__` seems clearer here

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.Focus.cs (hunk 3 above)
Comment (MartinZikmund) [reply src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.Focus.cs:2]: Whoopsie!

File: src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UWP/FeatureConfiguration/WinRTFeatureConfiguration.FileTypes.cs:48]: Fixed!

**Pull request merged**